### PR TITLE
Add block compression codec and remove not used codec

### PR DESCRIPTION
**Pull request by imay**

### CODE REVIEW

File: be/src/util/block_compression.cpp [hunk 1]
```diff
@@ -0,0 +1,387 @@
...
+    size_t _slice_off;
+};
+
+class SnappyBlockCompression : public BlockCompressionCodec {
+public:
+    static SnappyBlockCompression* instance() {
+        static SnappyBlockCompression s_instance;
+        return &s_instance;
+    }
+    ~SnappyBlockCompression() { }
```
Comment (kangpinghuang) [inline be/src/util/block_compression.cpp]: override or not?

**Review by kangpinghuang — DISMISSED** [review]

File: be/src/util/block_compression.cpp (hunk 1 above)
Comment (imay) [reply be/src/util/block_compression.cpp]: OK

File: gensrc/proto/segment_v2.proto [hunk 2]
```diff
@@ -61,8 +61,9 @@ enum CompressionTypePB {
     NO_COMPRESSION = 2;
     SNAPPY = 3;
     LZ4 = 4;
-    ZLIB = 5;
-    ZSTB = 6;
+    LZ4F = 5;
+    ZLIB = 6;
+    ZSTB = 7;
```
Comment (gaodayue) [inline gensrc/proto/segment_v2.proto]: typo? Does it mean [ZSTD](https://github.com/facebook/zstd) from facebook?

Comment (imay) [reply gensrc/proto/segment_v2.proto]: Yes, you are right. I'll fix it

**Review by chaoyli — APPROVED** [review]

**Pull request merged**